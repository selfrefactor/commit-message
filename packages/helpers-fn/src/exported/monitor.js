const { delay } = require('rambdax')
const { ms } = require('string-fn')

class Monitor{
  constructor(seconds = 5){
    this.highestMemoryUsage = undefined
    this.highestProcessUsage = undefined
    this.cycles = []
    this.stopFlag = false
    this.tick = ms(`${seconds} seconds`)
  }
  async start(){
    while(!this.stopFlag){
      await Promise.all([
        this.onEveryTick(),
        delay(this.tick)
      ])
    }
  }

  async onEveryTick(){
    console.log('foo')
  }
  async stopMonitor(){
    this.stopFlag = true
    await delay(this.tick)
    return {
      highestProcessUsage: this.highestProcessUsage,
      highestMemoryUsage: this.highestMemoryUsage,
      averageMemoryUsage: undefined,
      averageProcessUsage: undefined
    }
  }
  async stop(){
    const monitorData = await this.stopMonitor()
    console.log({monitorData})
  }
}

exports.Monitor = Monitor
exports.monitor = new Monitor()