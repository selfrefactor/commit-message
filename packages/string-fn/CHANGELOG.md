## Changelog

3.0.0

- Change licence from `ISC` to `MIT`

- Upgrade `rollup` setup

- Add `msToText`

2.7.0 Support extra latin

2.6.0 Add `fitWithinLines`

2.4.0 Add `splitPerLine`