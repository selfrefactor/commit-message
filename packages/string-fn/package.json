{
	"name": "string-fn",
	"version": "3.0.2",
	"typings": "./index.d.ts",
	"description": "String manipulation library",
	"main": "./dist/string-fn",
	"module": "./dist/string-fn.esm.js",
	"scripts": {
		"out": "yarn build:main&&yarn build:web",
		"build:main": "NODE_ENV=build rollup -c rollup.config.js",
		"build:web": "NODE_ENV=build rollup -c rollup.web.config.js",
		"test": "jest"
	},
	"files": [
		"dist",
		"index.d.ts"
	],
	"repository": "https://github.com/selfrefactor/services/tree/master/packages/string-fn",
	"keywords": [
		"string",
		"util",
		"words",
		"sentence",
		"manipulation"
	],
	"author": "self_refactor",
	"license": "MIT",
	"devDependencies": {
		"@babel/core": "7.12.3",
		"@babel/plugin-proposal-object-rest-spread": "7.12.1",
		"@babel/preset-env": "7.12.1",
		"@types/jest": "26.0.15",
		"babel-jest": "26.6.1",
		"jest": "26.6.1",
		"@rollup/plugin-babel": "5.2.1",
		"@rollup/plugin-commonjs": "15.1.0",
		"@rollup/plugin-json": "4.1.0",
		"@rollup/plugin-node-resolve": "9.0.0",
		"@rollup/plugin-replace": "2.3.3",
		"rollup": "2.32.1",
		"rollup-plugin-cleanup": "3.2.1",
		"rollup-plugin-sourcemaps": "0.6.3",
		"typescript": "4.1.0-dev.20201023"
	},
	"dependencies": {
		"rambdax": "7.0.1"
	}
}
