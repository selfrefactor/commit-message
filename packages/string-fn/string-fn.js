export * from './src/between'
export * from './src/camelCase'
export * from './src/count'
export * from './src/constantCase'
export * from './src/distance'
export * from './src/distanceGerman'
export * from './src/dotCase'
export * from './src/glob'
export * from './src/getIndent'
export * from './src/indent'
export * from './src/isLetter'
export * from './src/isPunctuation'
export * from './src/kebabCase'
export * from './src/maskSentence'
export * from './src/maskWords'
export * from './src/ms'
export * from './src/pascalCase'
export * from './src/removeIndent'
export * from './src/randomString'
export * from './src/reverse'
export * from './src/seoTitle'
export * from './src/shuffle'
export * from './src/snakeCase'
export * from './src/splitPerLine'
export * from './src/splitSentence'
export * from './src/stripPunctuation'
export * from './src/getMaxLength'
export * from './src/fitWithinLines'
export * from './src/stripTags'
export * from './src/takeArguments'
export * from './src/titleCase'
export * from './src/trim'
export * from './src/words'
export * from './src/wordsX'
