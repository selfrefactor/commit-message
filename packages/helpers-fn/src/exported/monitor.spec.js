import {monitor, getProcessUsage} from './monitor'
import { delay } from 'rambdax'
import { ms } from 'string-fn'

jest.setTimeout(ms('30 minutes'))

test.skip('getProcessUsage', async () => {
  const result = await getProcessUsage()
  console.log({result})
})

test('happy', async () => {
  monitor.start()
  await delay(30000)
  const logData = await monitor.stop()
  expect(
    logData
  ).toMatchSnapshot()
})