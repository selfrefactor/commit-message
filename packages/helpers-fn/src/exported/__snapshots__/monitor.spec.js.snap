// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Object {
  "avarages": Object {
    "memoryUsage": 5.7,
    "processUsage": 11.22,
  },
  "diff": Object {
    "memory": 0,
    "memoryAverage": 0,
    "process": -9.9,
    "processAvarage": -9.9,
  },
  "initial": Object {
    "memoryUsage": 5.7,
    "processUsage": 21.12,
  },
}
`;
